### PR TITLE
[NS-43][NS-48][NS-52] Finish new `VeProcess` integration

### PR DESCRIPTION
**Pull request by q10**
- Remove duplicated code associated with the new `VeProcess` integration efforts, now that the integration is complete
- Rebase `aveobench` subproject on the new `VeProcess` infrastructure
- Remove unused code in `VeProcessMetrics` now that most of the metrics tracking is integrated into `VeProcess` and `VectorEngine`
- Remove `PureVeFunctions` and its usage, as it is replaced by `SampleVeFunctions`
- Fix flaky tests that fail when  zero-sized sequences are generated
- Harden `VeProcess` methods to ensure there are no race conditions

### CODE REVIEW

File: src/main/scala/com/nec/vectorengine/VeProcessImpl.scala [hunk 1]
```diff
@@ -345,17 +347,19 @@ final case class WrappingVeo private (val node: Int,
...
     withVeoProc {
-      val npath = Paths.get(lib.path).normalize
-      loadedLibRecords.get(npath.toString) match {
-        case Some(lib) =>
-          logger.info(s"[${handle.address}] Unloading library from the VE process: ${npath}...")
-          val result = veo.veo_unload_library(handle, lib.value)
-          require(result == 0, s"Failed to unload library from the VE process, got ${result} (library at: ${npath})")
-          // Remove only after the veo_unload_library() was successful
-          loadedLibRecords.remove(npath.toString)
+      loadedLibRecords.synchronized {
```
Comment (treo) [inline src/main/scala/com/nec/vectorengine/VeProcessImpl.scala:350]: Do we need this synchronization now that `withVeoProc` has its own lock?

File: src/main/scala/com/nec/vectorengine/VeProcessImpl.scala [hunk 2]
```diff
@@ -406,15 +410,17 @@ final case class WrappingVeo private (val node: Int,
 
   def freeArgsStack(stack: VeCallArgsStack): Unit = {
     withVeoProc {
-      stackRecords.get(stack.args.address) match {
-        case Some(allocation) =>
-          logger.trace(s"[${handle.address}] Releasing veo_args @ ${stack.args.address}")
-          veo.veo_args_free(stack.args)
-          // Remove only after the free() was successful
-          stackRecords.remove(stack.args.address)
+      stackRecords.synchronized {
```
Comment (treo) [inline src/main/scala/com/nec/vectorengine/VeProcessImpl.scala:413]: Same question as above, do we need this additional lock?

File: src/main/scala/com/nec/vectorengine/VeProcessImpl.scala (hunk 1 above)
Comment (q10) [reply src/main/scala/com/nec/vectorengine/VeProcessImpl.scala:350]: The lock used by `withVeoProc` is meant to ensure that the process is not closed while a process method is executing - it allows multiple threads to check `open` state and use the process but only one thread can access the `open` when it is updated (i.e. during `process.close`).  This is separate from `loadedLibRecords.synchronized`, which is meant to enforce loading libraries one at a time.

File: src/main/scala/com/nec/vectorengine/VeProcessImpl.scala (hunk 2 above)
Comment (q10) [reply src/main/scala/com/nec/vectorengine/VeProcessImpl.scala:413]: Similar response to above.

**Pull request merged**